### ISSUE TITLE
[Feature Request]: VIsual indication of when the episode will count as 'watched'

### ISSUE THREAD

**Issue by xTheEc0** [ThaUnknown/miru#221]
### Preflight checklist

- [X] I have searched the [issue tracker](https://github.com/ThaUnknown/miru/issues) for a bug report that matches the one I want to file, without success.
- [X] I have searched the [features list](https://github.com/ThaUnknown/miru#features) for this feature, and I couldn't find it.
- [X] I have checked that I'm using the [latest stable](https://github.com/ThaUnknown/miru/releases/latest) version of the app.

### Problem Description

Some shows have a lot of buffer at the end, like previews for next episode, ED, netflix credits whatever.
After missing a few 'counts' of completed on some episodes I now instinctively jump to the last few seconds of the episode to make sure it counts when ED/credits start

### Proposed Solution

Some kind of visual indicator on the timeline (is that how the line where you can click to select what time to play called?)
Be it a red line, or anything else.
(Red line might scare people because red = danger, so I leave it up to your UX discretion)

**Comment by ThaUnknown** [comment]
I honestly never had this issue
TLDR the code is:
```js
const fromEnd = Math.max(180, duration / 10)
if (duration - fromEnd < currentTime) complete()
```
so 3 minutes or 1/10th the duration, whichever is bigger, from end is when the video gets marked as watched, everything I tried worked worse, like 75% total time watched etc

as Miru doesn't host any of it's content it has no clue as to what is playing, it just tries its best to guess based on the filename, as such it has no clue what's the outro or intro, it doesn't know if the video is playing credits etc so this solution by far worked best.

as someone who sometimes watches anime at x1.5 or constantly seeks 2 seconds ahead, I do not have any issues with the watched not being triggered, so don't get mad when I say "you're not watching your anime at all, you're barely skimming tru it", if however this isn't the case then the auto-completion needs to be re-considered a bit.

What the current approach considers is an user who wants to skip an ED which also might have its own wall of text credits which is getting somewhat common nowdays, which is why its 3 minutes. Usually if you watch the 1st ep you will know if the entire show has walls of text at the end, or if has aftercredits, so are you sure you're not just skipping aftercredits?

What u suggested is way too...... non-standard, so it will be insanely confusing for people which <2 braincells. Most of my UI/UX copies existing interfaces/standards so people don't have too much unique stuff to learn from, TLDR everything needs to be designed as if the person couldn't read and had really poor eyesight

**Comment by xTheEc0** [comment]
Makes sense, was just a suggestion : )
kek <2 braincells, truuue.

I'll just continue jumping to the very end as soon as ED/credits/whatever start : )

**Comment by ThaUnknown** [comment]
> Makes sense, was just a suggestion : ) kek <2 braincells, truuue.
> 
> I'll just continue jumping to the very end as soon as ED/credits/whatever start : )

you can just press S and that will be enough 90% of the time, but if it is a serious issue then my approach to this needs to be changed, so feel free to re-open the issue if its still annoying later on

**Comment by Eichen97** [comment]
would be cool to have a "mark as watched" button if an episode has some progress already
